cpp/src/io/parquet/page_decode.cuh

```diff
@@ -68,6 +69,12 @@ struct page_state_s {
   PageNestingDecodeInfo nesting_decode_cache[max_cacheable_nesting_decode_info];
   // points to either nesting_decode_cache above when possible, or to the global source otherwise
   PageNestingDecodeInfo* nesting_info;
+
+  inline __device__ void set_error_code(int32_t err) volatile
```

Comment:
Am I correct in assuming that different threads in a block may find different errors, and we set any one of the error codes here due to `memory_order_relaxed`? And then we extend this to `thread_scope_device` where different blocks may find different errors and we set the final error returned to the user as any one of them?

Which means errors are not deterministic?

Reply:
Not quite deterministic, yes. 
And this is regardless of the relaxed order, since we call three kernels with the same output error code address.
This may be a good argument to use a bitwise mask for the errors.

Reply:
How many items would the bitset report errors for? 1 bit per block or 1 bit per thread? Maybe we should also allow the user to "enable" host-side error reporting

Reply:
We wouldn't separate errors from different threads/blocks, just errors from different root causes but applying a bitwise OR when updating the error code. Final error code would represent all types of errors we encountered in any thread/block/kernel.

Reply:
Ah, yeah, we should definitely do this